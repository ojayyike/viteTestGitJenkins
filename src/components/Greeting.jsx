import { useState } from "react";

const Greeting = ({children}) => {
    const [changedText, setChangedText] = useState(false)

    const changeTextHandler = () => {
        setChangedText(!changedText)
    }

    return (
        <>
        <h2>Hello World</h2>
        {!changedText && <p>It is great to see you!</p>}
        {changedText && <p>Welcome to React</p>}
        <button data-testid="greeting-btn" onClick={changeTextHandler}>Change Text</button>
        </>
    
    )

}

export default  Greeting