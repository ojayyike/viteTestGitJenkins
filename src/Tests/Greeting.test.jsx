import Greeting from '../components/Greeting'
import { render,screen,fireEvent } from '@testing-library/react';
import renderer from 'react-test-renderer'
import '@testing-library/jest-dom'

describe('Greeting component', () => {
    test('renders "Hello World" as a text', () => {
        // Arrange
        render(<Greeting />)
        
        //Act
        // ...Nothing

        //Assert
        const helloWorldElement = screen.getByText(/hello world/i)
        expect(helloWorldElement).toBeInTheDocument();

    })

    test('renders great to see you when button NOT clicked', () => {
        // Arrange
        render(<Greeting />)
        
        //Act
        //..Nothing

        //Assert
        const outputElement = screen.getByText(/It is great to see you!/i)
        expect(outputElement).toBeInTheDocument();

    })

    test('renders Welome to react when button clicked', async () => {
        //Arrange
        render(<Greeting />);
 
        //act (perform the main action)
        await fireEvent.click(screen.getByTestId('greeting-btn'))
 
        //assert the results (see if they match our expectations)
        const outputChangedElement = screen.getByText('Welcome to React', { exact: false });
        expect(outputChangedElement).toBeInTheDocument();
    })

    test('does not render good to see you when button is clicked',async () => {
        //Arrange
        render(<Greeting />);

        await fireEvent.click(screen.getByTestId('greeting-btn'))


        //Assert 
        const outputElement = screen.queryByText(/good to see you/i)
        expect(outputElement).toBeNull()
    })
})

describe('Greeting component snapshot', () => {
    //Snapshot test to ensure UI component was not altered unexpectedly

    test("Ensure UI matches snapshot", () => {
        const tree = renderer.create(<Greeting />).toJSON()
        expect(tree).toMatchSnapshot();
    }) 
}
)

