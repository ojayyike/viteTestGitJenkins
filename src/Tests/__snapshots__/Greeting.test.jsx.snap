// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Greeting component snapshot Ensure UI matches snapshot 1`] = `
Array [
  <h2>
    Hello World
  </h2>,
  <p>
    It is great to see you!
  </p>,
  <button
    data-testid="greeting-btn"
    onClick={[Function]}
  >
    Change Text
  </button>,
]
`;
